src/library/scala/collection/Iterator.scala

```diff
@@ -212,44 +211,39 @@ trait Iterator[+A] extends IterableOnce[A] with IterableOnceOps[A, Iterator, Ite
       buf
     }
 
-    private def padding(x: Int) = immutable.ArraySeq.untagged.fill(x)(pad.get())
-    private def gap = (step - size) max 0
+    private def gap =
+      (step - size) max 0
 
     private def go(count: Int) = {
       val prevSize = buffer.size
       def isFirst = prevSize == 0
-      // If there is padding defined we insert it immediately
-      // so the rest of the code can be oblivious
-      val xs: Seq[B] = {
-        val res = takeDestructively(count)
-        // was: extra checks so we don't calculate length unless there's reason
-        // but since we took the group eagerly, just use the fast length
-        val shortBy = count - res.length
-        if (shortBy > 0 && pad.isDefined) res ++ padding(shortBy) else res
+      val xs = {
+        // If there is padding defined we insert it immediately
+        // so the rest of the code can be oblivious
+        val xs = takeDestructively(count)
```

Comment:
I do not like defining `xs` inside of `xs`. it's too easy to get confused

Reply:
Actually I don't think we need this nested block